cluster-autoscaler/update_toc.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
+
+# Copyright YEAR The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+faq_file = open("FAQ.md","r")
+faq_content = faq_file.read()
+faq_file.close()
+
+question_prefix = "### "
+questions = [line.strip()[len(question_prefix):].strip() for line in faq_content.split("\n") if line.strip().startswith(question_prefix)]
+faq_file = open("FAQ.md","w")
+in_toc = False
+
+for line in faq_content.split("\n"):
+    if line.strip() == "<!--- TOC BEGIN -->":
+	in_toc = True
+	faq_file.write(line +"\n")
+	for question in questions:
+	    faq_file.write("* [%s](#%s)\n" % (question, re.sub("[^a-z0-9 ]+", "", question.lower()).replace(" ","-")))
+    if line.strip() == "<!--- TOC END -->":
+	in_toc = False
+    if not in_toc:
+	faq_file.write(line+"\n")
+
+faq_file.close()
```

Comment:
My instinct would be to put the logic in a function, so it can be imported and just call this function in `if __name__ == '__main__'` block.